src/nvim/eval/funcs.c

```diff
@@ -8037,12 +8038,14 @@ static void f_screenattr(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 
   int row = (int)tv_get_number_chk(&argvars[0], NULL) - 1;
   int col = (int)tv_get_number_chk(&argvars[1], NULL) - 1;
-  if (row < 0 || row >= default_grid.rows
-      || col < 0 || col >= default_grid.cols) {
+
+  ScreenGrid *grid = ui_comp_get_grid_at_coord(row, col);
+  screenchar_adjust_coords(&grid, &row, &col);
```

Comment:
I'd suggest move the call to ui_comp_get_grid_at_coord inside `screenchar_adjust_coords`.

Reply:
Done.